src/dap/request_handler.py

```diff
@@ -168,6 +168,128 @@ def _attach(self, request: dict) -> Iterator[dict]:
                 hit_breakpoint_ids=[],
             )
 
+    @register_command("listProcesses")
+    def _list_processes(self, request: dict) -> Iterator[dict]:
+        """Process the command `listProcesses`."""
+        success, message, processes = self.gdb_backend.process_manager.get_processes()
+        current_pid = self.gdb_backend.process_manager.get_current_pid()
+
+        response = {
+            "type": "response",
+            "request_seq": request.get("seq", 0),
+            "command": "listProcesses",
+            "success": success,
+            "message": message,
+            "body": {"processes": processes, "currentProcess": current_pid},
+        }
+        yield response
+
+    @register_command("addInferiors")
+    def _add_inferiors(self, request: dict) -> Iterator[dict]:
+        """Process the command `addInferiors`."""
+        pids = self._get_argument(request, "pids", [])
+        if self.notifier:
+            # When adding inferiors, gdb briefly starts and stops the program being debugged.
+            # The client does not need to know about this
+            self.notifier.stop_notifier()
```

Comment:
NIT
i would strongly recommend to use a context manager for this kind of things

```
with self.without_notifications():
   self.gdb_backend.process_manager.add_inferior_with_pids(pids)
```
or something like it

Reply:
nit #2

you also can move the `if self.notifier` check in this context manager